archive_repo.py

```diff
@@ -0,0 +1,667 @@
+            ...commentFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_Comment_Fields
+    + gql_AuthorFields
+)
+
+gql_PullRequest_Fields = (
+    """
+fragment prFields on PullRequest {
+    number
+    id
+    title
+    url
+    state
+    ...labels
+    ...author
+    ...commentFields
+    closedAt
+    mergedAt
+    mergedBy { ...userFields }
+    comments(first: 100) {
+        nodes {
+            ...author
+            ...commentFields }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+    reviews(first: 50) {
+        nodes {
+            ...reviewFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_LabelFields
+    + gql_Review_Fields
+)
+# ...reviewFields definition includes ...commentFields and ...author
+# ...author definition includes ...userFields
+
+gql_PullRequest_Query = (
+    """
+      nodes { ...prFields }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+      totalCount
+    }
+  }
+  rateLimit {
+    remaining
+    resetAt
+  }
+}
+"""
+    + gql_PullRequest_Fields
+)
+
+gql_AllPRs_Initial = (
+    """
+query($owner: String!, $repo: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 10, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_AllPRs_Subsequent = (
+    """
+query($owner: String!, $repo: String!, $cursor: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 25, after: $cursor, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_PR_Comments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            comments(first:100, after:$cursor) {
+                nodes { ...commentFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+gql_PR_Review_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            reviews(first:100, after:$cursor) {
+                nodes { ...reviewFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Review_Fields
+)
+
+gql_PR_ReviewComments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequestReview {
+            comments(first: 50, after:$cursor) {
+                nodes {
+                    originalPosition
```

Comment:
Are you sure that we want `originalPosition` and not `position`?

Reply:
Frankly, I'm not sufficiently clear on the difference.  What I'd really like is a line number in the commit, rather than the line number in the diff (original or not), but I don't see that in the API.

Reply:
Yeah, I saw that; it's not very nice.  I suspect that this is a feature that GH is stuck with.  It's probably why you can't comment on lines that aren't in the diff as well.  But maybe they will eventually be able to fix it, who knows.